libnetwork/internal/netiputil/netiputil.go

```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/docker/docker/libnetwork/ipbits"
 )
 
-func toIPNet(p netip.Prefix) *net.IPNet {
+func ToIPNet(p netip.Prefix) *net.IPNet {
 	if !p.IsValid() {
 		return nil
 	}
```

Comment:
Not new code, but looks like this is now exported, and yet-another potential foot-gun. This function looks to be silently discarding invalid values, but doesn't describe that. We should (can be a follow-up) at least document the behavior

But also curious if we would benefit from this function returning an error (instead of just a `nil` value). I see we're using this function here;

```go
if err := i.nlh.AddrReplace(i.Link, &netlink.Addr{IPNet: netiputil.ToIPNet(addrPrefix)}); err != nil {
```

Which would mean we're potentially passing a `nil` `IPNet`;

- is `addrPrefix` already validated before we reach that code?
- or is `nil` a valid value for that call?
- if not; are we wasting potential cycles here (calling whatever `i.nlh.AddrReplace` does)?

If none of the above; should we check `addrPrefix.IsValid()` before we make that call, and skip  `i.nlh.AddrReplace` ? In that case, do we need an error (or a log)?

Reply:
It's already validated (either the link local address, or from an IPAM pool).

As this is a low-level util, used for already-validated addresses, I think returning an error would just clutter the calling code.

So, I've added a comment noting the `nil` return on error.

Reply:
Thanks for adding the comment; I indeed expected that validation already happened before this.

We should look more at moving utilities like this to `internal` packages if they  are potential foot-guns for those unaware of its behavior. Then again, we probably want "all of libnetwork" to be mostly internal, because there's a lot of code not well-designed for "general use".

Reply:
And I just realised it's "internal" 😂 (missed the file rename / move, but now saw it when looking at individual commits) my bad!